packages/testcontainers/src/wait-strategies/one-shot-startup-startegy.test.ts

```diff
@@ -0,0 +1,31 @@
+import { GenericContainer } from "../generic-container/generic-container";
+import { Wait } from "./wait";
+import { checkContainerIsHealthy } from "../utils/test-helper";
+
+jest.setTimeout(180_000);
+
+const mockImageInspect = jest.fn();
```

Comment:
The reason why the tests are failing is because you are mocking the entirety of Dockerode, so Testcontainers ends up failing to create a valid container runtime strategy. I guess you only want to mock the image inspect function

Reply:
Even better, is it really necessary to mock anything here? I think we can just start a container that exits immediately

Reply:
Ah yes this was it! Thank you :) I added basic test, but I am still thinking how I test failed cases

Reply:
fixed, do you want me to squash my commits?

Reply:
Thanks. No need I'll squash the commits on merge